lib/instrumentation/generic-span.js

```diff
@@ -83,16 +83,6 @@ Object.defineProperty(GenericSpan.prototype, 'traceparent', {
   }
 })
 
-// duration is returned as an int representing the time in milliseconds
-GenericSpan.prototype.duration = function () {
-  if (!this.ended) {
-    this._agent.logger.debug('tried to call duration() on un-ended transaction/span %o', { id: this.id, parent: this.parentId, trace: this.traceId, name: this.name, type: this.type })
-    return null
-  }
-
-  return this._timer.duration
-}
```

Comment:
REVIEW NOTE: Having this `.duration()` be a lazy fetch on `this._timer` means that we could not remove the `this._timer` reference on `Transaction#end()` and `Span#end()` to free up some memory. I instead changed this to be a `duration` value on the Transaction and Span objects.  The `.duration()` is API is not documented. However, Alan, if you feel this is a breaking change, I can change this implementation to still have a `.duration()` method. Let me know.